R/render.R

```diff
@@ -325,6 +330,13 @@ render <- function(input,
       rmarkdown.runtime = runtime
     )
 
+    # read root directory from argument (has precedence) or front matter
+    root_dir <- knit_root_dir
```

Comment:
They are already supported -- i.e. this works:

    ---
    output: html_document
    knit_root_dir: data
    ---

In the YAML metadata, knit_root_dir is considered to be relative to the document directory. Do you think it should be relative to another reference point?